### PR TITLE
Update protobuf version to v3.0.0-beta-2

### PR DESCRIPTION
**Pull request by pszemus**
Commit da717f464d667aca410f3a0ddeaa7ab45d34b7d3 introduced new protobuf API usage that is available since https://github.com/google/protobuf/tree/v3.0.0-beta-2 (API changes here: https://github.com/google/protobuf/commit/a6361a124a6537c69aaaa4e667372ba4e89c1fd1)

### CODE REVIEW

**Comment by murgatroid99** [comment]
This is ok to test

**Comment by pszemus** [comment]
What's wrong with those tests?

**Comment by jcanizales** [comment]
CC @jtattermusch

**Comment by jcanizales** [comment]
I've restarted the tests, as the failures seem unrelated to this.

File: examples/cpp/route_guide/Makefile [hunk 1]
```diff
@@ -86,7 +86,7 @@ ifneq ($(HAS_VALID_PROTOC),true)
 	@echo "Please install Google protocol buffers 3.0.0 and its compiler."
 	@echo "You can find it here:"
 	@echo
-	@echo "   https://github.com/google/protobuf/releases/tag/v3.0.0-alpha-1"
+	@echo "   https://github.com/google/protobuf/releases/tag/v3.0.0-beta-2"
```
Comment (jtattermusch) [inline examples/cpp/route_guide/Makefile]: Since this file has been updated, please update it's copyright to 2015-2016, otherwise the sanity test will fail:
`examples/cpp/route_guide/Makefile modified 2016 but copyright only extends to 2015`

Same for helloworld.

**Comment by jtattermusch** [comment]
Interesting, I thought I've done all the necessary changes in https://github.com/grpc/grpc/pull/4787.

Thanks for the PR.

File: examples/cpp/route_guide/Makefile (hunk 1 above)
Comment (jtattermusch) [inline examples/cpp/route_guide/Makefile]: Friendly ping, will you be able to update the copyrights?

Comment (pszemus) [inline examples/cpp/route_guide/Makefile]: Copyrights are not related to this commit. It should be done in another PR and by someone else.

Comment (jtattermusch) [inline examples/cpp/route_guide/Makefile]: They are related. You are modifying the files examples/cpp/helloworld/Makefile and examples/cpp/routeguide/Makefile, and which also requires updating the copyright, otherwise our test suite checking the copyright will fail.
If you can't update the copyright, I won't be able to merge this PR - can't afford to break the tests (and I will probably need to go a more complicated way to get these changes in).

Comment (pszemus) [inline examples/cpp/route_guide/Makefile]: OK, added https://github.com/grpc/grpc/commit/fbf8c0b8ca235e606dcb914379825f6ff2691b7b to this PR
